src/components/VmDetails/cards/OverviewCard/index.js

```diff
@@ -88,16 +112,21 @@ class OverviewCard extends React.Component {
       this.props.onEditChange(true, true)
     }
 
+    const newState = { isDirty: true }
     let updates = this.state.vm
     // NOTE: The DetailsCard has the possibility of chained updates.  Overview doesn't
     //       have that need, so there is no __changeQueue__ setup here.
 
     let fieldUpdated
     switch (fieldName) {
       case 'name':
-        // TODO: add name validation?
         updates = updates.set('name', value)
         fieldUpdated = 'name'
+        newState.nameError = !isVmNameValid(value)
+        newState.disableHostnameToggle = !isHostNameValid(value)
+        if (newState.disableHostnameToggle) {
+          newState.updateCloudInit = false
```

Comment:
Cool already added the host name validation function (with its different requirements from the VM name).  Now I suggest still showing the checkbox (but disabled) and a different error message describing that the VM name cannot be used as the OS hostname.

@gregsheremeta , @lwrigh - WDYT?

Reply:
makes sense